src/test/app/Credentials_test.cpp

```diff
@@ -0,0 +1,949 @@
+                    ter(tecNO_PERMISSION));
+                env.close();
+            }
+
+            {
+                testcase("CredentialsDelete fail, no Issuer and Subject.");
+
+                auto jv =
+                    credentials::deleteCred(subject, subject, issuer, credType);
+                jv.removeMember(jss::Subject);
+                jv.removeMember(jss::Issuer);
+                env(jv, ter(temMALFORMED));
+                env.close();
+            }
+
+            {
+                testcase("CredentialsDelete fail, invalid fee.");
+
+                auto jv =
+                    credentials::deleteCred(subject, subject, issuer, credType);
+                jv[jss::Fee] = -1;
+                env(jv, ter(temBAD_FEE));
+                env.close();
+            }
+
+            {
+                testcase("deleteSLE fail, bad SLE.");
+                auto view = std::make_shared<ApplyViewImpl>(
+                    env.current().get(), ApplyFlags::tapNONE);
+                auto ter =
+                    ripple::credentials::deleteSLE(*view, {}, env.journal);
+                BEAST_EXPECT(ter == tecNO_ENTRY);
+            }
+        }
+    }
+
+    void
+    testFeatureFailed(FeatureBitset features)
+    {
+        using namespace test::jtx;
+
+        const char credType[] = "abcde";
+        Account const issuer{"issuer"};
+        Account const subject{"subject"};
+
+        {
+            using namespace jtx;
+            Env env{*this, features};
+
+            env.fund(XRP(5000), subject, issuer);
+            env.close();
+
+            {
+                testcase("Credentials fail, Feature is not enabled.");
+                env(credentials::create(subject, issuer, credType),
+                    ter(temDISABLED));
+                env(credentials::accept(subject, issuer, credType),
+                    ter(temDISABLED));
+                env(credentials::deleteCred(subject, subject, issuer, credType),
+                    ter(temDISABLED));
+            }
+        }
+    }
+
+    void
+    testRPC()
+    {
+        using namespace test::jtx;
+
+        const char credType[] = "abcde";
+        Account const issuer{"issuer"};
+        Account const subject{"subject"};
+
+        {
+            using namespace jtx;
+            Env env{*this};
+
+            env.fund(XRP(5000), subject, issuer);
+            env.close();
+
+            env(credentials::create(subject, issuer, credType));
+            env.close();
+
+            env(credentials::accept(subject, issuer, credType));
+            env.close();
+
+            testcase("account_tx");
+
+            std::string txHash0, txHash1;
+            {
+                Json::Value params;
+                params[jss::account] = subject.human();
+                auto const jv = env.rpc(
+                    "json", "account_tx", to_string(params))[jss::result];
+
+                BEAST_EXPECT(jv[jss::transactions].size() == 4);
```

Comment:
Why are there 4 transactions? Shouldn't it be the funding transaction, the `CredentialCreate`, and the `CredentialAccept`?

Reply:
+accountSet